Gallery-Server/src/main/java/edu/yu/cs/gallery/ServerStartUp.java

```diff
@@ -0,0 +1,50 @@
+package edu.yu.cs.gallery;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.springframework.web.reactive.function.client.WebClient;
+
+import edu.yu.cs.gallery.repositories.GalleryRepository;
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.runtime.QuarkusApplication;
+import io.quarkus.runtime.StartupEvent;
+import io.quarkus.runtime.annotations.QuarkusMain;
+import reactor.core.publisher.Mono;
+
+public class ServerStartUp {
+
+    @Inject
+    GalleryRepository gr;
+    
+    static Gallery gallery;
+
+    @Transactional
+    public void initEagerly(@Observes StartupEvent se) throws MalformedURLException {
+        gallery = gr.findAll().firstResult();
```

Comment:
This is the third time this bit is duplicated. As above, I believe it should be placed in a separate class which is injected into this class.